main.c

```diff
@@ -584,55 +580,76 @@ mtapPortOpen(3); >> Memory Card Port 2 (Logical MC Port 2)
  2D: PORT = 1,SLOT = 3
 ====================== 
  */
+ 
+// Detects If Multi-tap is Connected. Closes Multi-tap Port(s) if Multi-tap(s) are not Connected
+void mtapDetect()
 {
-	closeMTAPports(); //In Case Multi-tap(s) are Connected or removed After the App has been started.
-	int rv;
-	mtapPortOpen(2);
-	mtapPortOpen(3);
-
-	scr_printf("Multi-tap Status: \n");
-	//Checks For Mtap Connection on Physical Memory Card Slot 1
-	rv = mtapGetConnection(2); // Checks MTAP port 2 (Memory Card Port 1) For MTAP Connection
-	
-	if(rv == 1)
-	{
-		scr_printf("Memory Card Port 0: Multi-tap Detected! \n");
-	}
-	else
+    int mt;
+    scr_printf("Multi-tap Status: \n");
+    mtapPortOpen(2); // Checks MTAP port 2 (Memory Card Port 1) For MTAP Connection
+    mt = mtapGetConnection(2);
+    if(mt == 1)
     {
-		scr_printf("Memory Card Port 0: Multi-tap is Not Connected. \n");
-		mtapPortClose(2); //Closes The Multitap Port if The Multitap Is Not Present
+        scr_printf("Memory Card Port 0: Multi-tap Detected! \n");
+    }
+    else
+    {
+        scr_printf("Memory Card Port 0: Multi-tap is Not Connected. \n");
+        mtapPortClose(2); //Closes The Multitap Port if The Multitap Is Not Present
     }
-	
-	
-	//Checks For Mtap Connection on Physical Memory Card Slot 2
-	rv = mtapGetConnection(3); // Checks MTAP port 3 (Memory Card Port 2) For MTAP Connection
     
-	if(rv == 1)
-	{
-		scr_printf("Memory Card Port 1: Multi-tap Detected! \n");
-	}
-	else
+    mtapPortOpen(3);
+    mt = mtapGetConnection(3); // Checks MTAP port 3 (Memory Card Port 2) For MTAP Connection
+    if(mt == 1)
     {
-		scr_printf("Memory Card Port 1: Multi-tap is Not Connected. \n");
-		mtapPortClose(3);
-	}
-}
-// Closes MTAP Ports 2 and 3
-void closeMTAPports()
-{
-	mtapPortClose(2);
-	mtapPortClose(3);
+        scr_printf("Memory Card Port 1: Multi-tap Detected! \n");
+    }
+    else
+    {
+        scr_printf("Memory Card Port 1: Multi-tap is Not Connected. \n");
+        mtapPortClose(3); //Closes The Multitap Port if The Multitap Is Not Present
+    }
 }
-void gotoOSDSYS(int failure)
+
+void gotoOSDSYS(int sc)
 {
-	if (failure == 1)
+	if (sc == 1)
+	{
+		scr_printf(appFail);
```

Comment:
Since no matter what you are going to be printing appFail, you can move it out of the if statement and only have it written once. The same thing could probably be done with modloadFail if it wasn't for the fact that status 6 exists. You might be able to get away with doing it once and doing something like:

if(sc==1 || sc ==2 || etc)
{
scr_printf(modloadfail);
}